Adds environment variables for all proxy settings

jaffinito opened the pull request:
## Description

* Adds environment variables for all proxy settings
* Updates string version of EnvironmentOverrides to properly handle whitespace
* Adds unit tests for each proxy setting, testing local and env vars
* change log updates!!!!!!

Closes #878

# Author Checklist
- [x] Unit tests, Integration tests, and Unbounded tests completed
- [x] Performance testing completed with satisfactory results (if required)
- [x] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

JcolemanNR commented on src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs:
```diff
@@ -1984,11 +2019,18 @@ private List<string> EnvironmentOverrides(List<string> local, params string[] en
+                .FirstOrDefault();
+
+            // prevent whitespace from leaking through
+            if (string.IsNullOrWhiteSpace(envValue))
```
Thoughts/feelings about trimming the value we read? I can't think of any use cases where we'd want extra whitespace at the start/end of env vars

jaffinito replied:
Yeah, I was waffling back and forth on using trim vs just detecting.  I am fine using trim, what do you think?

nr-ahemsath replied:
I think trimming makes sense.

JcolemanNR replied:
I agree, that we should trim this. I can only see it helping customers

JcolemanNR reviewed (commented):
These changes look good. Please see/respond to my open question about trimming values we read from env vars... I'm just submitting a non-approving review for now

nr-ahemsath commented on src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs:
```diff
@@ -1332,15 +1360,16 @@ public virtual string ProxyPassword
                 if (!_proxyPasswordEvaluated)
                 {
                     var hasObscuringKey = !string.IsNullOrWhiteSpace(ObscuringKey);
```
It's worth noting, I think, that the obscuring key can also be configured via the env var `NEW_RELIC_CONFIG_OBSCURING_KEY`.  In our config docs, we currently tell users to put the obscuring key in the environment and the obscured values in the config file (otherwise the obfuscation is almost pointless).   We might need to update that statement when we update the config docs with these new env vars.

jaffinito replied:
Yeah, I will leave this here so I remember when working on docs

JcolemanNR reviewed (approved):
Looks good!

jaffinito commented:
Updated docs branch (https://github.com/jaffinito/docs-website/tree/dotnet-grass-valley-feature-pack) with new env vars and updated the obscuring portion to better reflect the new settings.
